collect_app/src/androidTest/java/org/odk/collect/android/support/pages/Page.kt

```diff
@@ -0,0 +1,409 @@
+package org.odk.collect.android.support.pages
+
+import android.app.Activity
+import android.content.pm.ActivityInfo
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.NoMatchingViewException
+import androidx.test.espresso.PerformException
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.assertion.ViewAssertions
+import androidx.test.espresso.contrib.RecyclerViewActions
+import androidx.test.espresso.core.internal.deps.guava.collect.Iterables
+import androidx.test.espresso.matcher.RootMatchers
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.runner.lifecycle.ActivityLifecycleMonitorRegistry
+import androidx.test.runner.lifecycle.Stage
+import org.hamcrest.CoreMatchers
+import org.hamcrest.Matchers
+import org.hamcrest.core.StringContains
+import org.hamcrest.core.StringEndsWith
+import org.junit.Assert
+import org.odk.collect.android.R
+import org.odk.collect.android.storage.StoragePathProvider
+import org.odk.collect.android.support.AdbFormLoadingUtils
+import org.odk.collect.android.support.CustomMatchers
+import org.odk.collect.android.support.WaitFor.wait250ms
+import org.odk.collect.android.support.WaitFor.waitFor
+import org.odk.collect.android.support.actions.RotateAction
+import org.odk.collect.android.support.matchers.RecyclerViewMatcher
+import org.odk.collect.android.utilities.TranslationHandler
+import org.odk.collect.androidshared.ui.ToastUtils.popRecordedToasts
+import org.odk.collect.testshared.NestedScrollToAction
+import timber.log.Timber
+import java.io.File
+import java.io.IOException
+
+/**
+ * Base class for Page Objects used in Espresso tests. Provides shared helpers/setup.
+ *
+ *
+ * Sub classes of `Page` should represent a page or part of a "page" that the user would
+ * interact with. Operations on these objects should return `this` (unless
+ * transitioning to a new page) so that they can be used in a fluent style.
+ *
+ *
+ * The generic typing is a little strange here but enables shared methods such as
+ * [Page.closeSoftKeyboard] to return the sub class type rather than `Page` so
+ * operations can be chained without casting. For example `FooPage` would extend
+ * `Page<FooPage>` and calling `fooPage.closeSoftKeyboard()` would return
+ * a `FooPage`.
+ *
+ * @see [Page Objects](https://www.martinfowler.com/bliki/PageObject.html)
+ *
+ * @see [Fluent Interfaces](https://en.wikipedia.org/wiki/Fluent_interface)
+ */
+
+abstract class Page<T : Page<T>> {
```

Comment:
Across the codebase we try to import static methods like:
- onView
- withText
- matches

etc so that instead of for example: `Espresso.onView` we just call `onView` what should make tests a bit more readable. Please take a look at other tests and clean this one.

Reply:
Done.